src/rt/sched/status.h

```diff
@@ -0,0 +1,123 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+#include "../ds/morebits.h"
+
+#include <ostream>
+
+namespace verona::rt
+{
+  /// TODO: backpressure docs
+  enum struct BackpressureState : uint8_t
+  {
+    Normal = 0b00,
+    Muted = 0b01,
+    Unmutable = 0b10,
+    MaybeUnmutable = 0b11,
+
+    IsUnmutable = 0b10,
+  };
+
+  inline bool operator&(BackpressureState s1, BackpressureState s2)
+  {
+    return (uint8_t)s1 & (uint8_t)s2;
+  }
+
+  inline std::ostream& operator<<(std::ostream& os, BackpressureState s)
+  {
+    switch (s)
+    {
+      case BackpressureState::Normal:
+        return os << "Normal";
+      case BackpressureState::Muted:
+        return os << "Muted";
+      case BackpressureState::Unmutable:
+        return os << "Unmutable";
+      case BackpressureState::MaybeUnmutable:
+        return os << "MaybeUnmutable";
+      default:
+        return os << "??";
```

Comment:
According to the enum, this would be `IsUnmutable`. Why `??`?

Reply:
Now I get it, `IsImutable` is just a flag test and not an actual value. This is confusing, as I explained below. The fact that this isn't an assert adds up to the confusion with the lack of documentation and opaque implementation.